Mathlib/LinearAlgebra/AffineSpace/Independent.lean

```diff
@@ -866,6 +866,16 @@ def faceOpposite {n : ℕ} [NeZero n] (s : Simplex k P n) (i : Fin (n + 1)) : Si
     Set.range (s.faceOpposite i).points = s.points '' {i}ᶜ  := by
   simp [faceOpposite]
 
+lemma faceOpposite_point_eq_point (s : Simplex k P 1) {i j : Fin 2} (h : i ≠ j) (n : Fin 1) :
+    (s.faceOpposite i).points n = s.points j := by
```

Comment:
What do you think of
```suggestion
    (s.faceOpposite i).points n = s.points I.rev := by
```
which could be a simp lemma?

Reply:
I've changed the statement to a `Fin.rev` one as that seems convenient for the 1-simplex case, but there's still the matter of the more general `succAbove` form.

Reply:
Though actually I think it suffices to state this lemma for an index type of the form

Reply:
Well, you could make `faceOpposite` take an additional argument for definitional control of the type of the result rather than forcing it to involve `n - 1`, and so keep it applying to any `Simplex k P n` with `[NeZero n]` rather than only those that are of a type defeq to `Simplex k P (n + 1)`, but such an additional argument would be irrelevant to almost all uses that don't actually care about the precise indices in `faceOpposite` (but only e.g. projections onto its span).

Reply:
I've added the `succAbove` statement and used it to prove the `Fin.rev` one. It's possible there is some more generally applicable lemma about `orderEmbOfFin` for the complement of a singleton and its relation to `succAbove` that should be extracted.

Reply:
I think probably that lemma should be stated without `h'`, but otherwise that looks like a good lemma.

Reply:
OK, added that lemma (plus an applied version).